### ISSUE TITLE
Set the unhandled/handled state

### ISSUE THREAD

**Issue by sandstrom** [bugsnag/bugsnag-ruby#549]
### Description
Currently there is no easy way to set the handled/unhandled state for an exception.

In our Rails application we are catching all exceptions using `rescue_from`, to render an error json payload + send it to Bugsnag. While we do "handle" the exception, in the sense that we're catching it. The resulting status code will still be `500` and the user will see an error message.

So, from a stability perspective it wasn't handled. Thus, we'd like these errors to worsen the Bugsnag stability score. And for that to work we'd like to set the handled/unhandled state via `Bugsnag.notify`.

```ruby
rescue_from StandardError, :with => :custom_error_render
```

### Issue

We'd like more control over the handled/unhandled state. For example by making the property adjustable on the report instance.

### Environment

Library versions: 6.11.1

**Comment by tobyhs** [comment]
Hi @sandstrom 

I don't think bugsnag-ruby has an officially documented way to send unhandled errors, but providing a second argument (auto_notify) of `true` for `Bugsnag.notify` (like `Bugsnag.notify(exception, true)`) should send an unhandled error. Does this work for your use case?

**Comment by sandstrom** [comment]
@tobyhs Yeah I noticed that in the code, but that doesn't really do what I want.

Auto-notify is orthogonal from handled/unhandled. More specifically, setting auto_notify will have other implications, besides flagging the exception has unhandled.

https://github.com/bugsnag/bugsnag-ruby/blob/master/lib/bugsnag.rb#L72-L73

I've added a small PR that illustrate how easy (I think) this fix should be:
https://github.com/bugsnag/bugsnag-ruby/pull/550

**Comment by tobyhs** [comment]
For being able to set unhandled on an error report, we have an existing internal roadmap ticket for this as this has come up for a few other notifier libraries recently.

We would need to come up with and decide on an approach/design that would likely be consistent across the languages/platforms that Bugsnag supports. We can consider your pull request when the approach has been decided on.

**Comment by sandstrom** [comment]
Thanks @tobyhs! Looking forward to it! Any tentative ETA?

**Comment by tobyhs** [comment]
We don't have an ETA at the moment.

**Comment by sandstrom** [comment]
@phillipsam Any update on this?

I read about the new user stability feature (which looks great! 😄). But we can't use it because of the problem described in this issue.

**Comment by johnkiely1** [comment]
Hi @sandstrom,

Nothing new to report on timeframes on this yet. We are still looking to address this across all of our platform specific notifiers which makes this a bigger piece of work to implement. Other priorities have meant this has slipped somewhat. Updates will be added here when we have them.

**Comment by Drew-Goddyn** [comment]
Also interested in seeing progress on this.

**Comment by sandstrom** [comment]
Hello @johnkiely1, just wanted to check-in and see if there is any progress or ETA on this? 😄 

When this finally shipped for JS client last year we could finally start using the stability dashboard, which has been great.

**Comment by yousif-bugsnag** [comment]
Hi @sandstrom,

There's still no ETA for this yet I'm afraid. It is still on our product roadmap, and we're still planning to add support for this when priorities allow. We'll be sure to let you know of any updates.

**Comment by sandstrom** [comment]
Thanks for the update @yousif-bugsnag 

As for the problem, anyone using `rescue_from` (https://edgeapi.rubyonrails.org/classes/ActiveSupport/Rescuable/ClassMethods.html), a common Rails functionality, to manually render an error page, etc. will have all their errors marked as handled and have 0% use of your stability tracking.

Which is a pretty big drawback.

Just wanted to explain in case it helps you prioritize! 😄
